x/evmutils/keeper/bank_keeper.go

```diff
@@ -0,0 +1,209 @@
+package keeper
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+
+	"github.com/kava-labs/kava/x/evmutils/types"
+)
+
+const (
+	// EvmDenom is the gas denom used by evm
+	EvmDenom = "akava"
+
+	// CosmosDenom is the gas denom used by the kava app
+	CosmosDenom = "ukava"
+)
+
+// ConversionMultiplier is the conversion multiplier between akava and ukava
+var ConversionMultiplier = sdk.NewInt(1_000_000_000_000)
+
+// EvmBankKeeper is a BankKeeper wrapper for the x/evm module to allow the use
+// of the 18 decimal akava coin on the evm.
+// This keeper uses both the ukava coin and a separate akava coin to
+// manage the extra percision needed by the evm.
+type EvmBankKeeper struct {
+	bk types.BankKeeper
+}
+
+func NewEvmBankKeeper(bk types.BankKeeper) EvmBankKeeper {
+	return EvmBankKeeper{
+		bk: bk,
+	}
+}
+
+// GetBalance returns the total **spendable** balance of akava for a given account by address.
+func (k EvmBankKeeper) GetBalance(ctx sdk.Context, addr sdk.AccAddress, denom string) sdk.Coin {
+	if denom != EvmDenom {
+		panic(fmt.Errorf("only evm denom %s is supported by EvmBankKeeper", EvmDenom))
+	}
+
+	spendableCoins := k.bk.SpendableCoins(ctx, addr)
+	ukava := spendableCoins.AmountOf(CosmosDenom)
+	akava := spendableCoins.AmountOf(EvmDenom)
+	total := ukava.Mul(ConversionMultiplier).Add(akava)
+	return sdk.NewCoin(EvmDenom, total)
+}
+
+// SendCoinsFromModuleToAccount transfers 18 decimal coins from a ModuleAccount
+// to an AccAddress. It will panic if the module account does not exist. An
+// error is returned if the recipient address is black-listed or if sending the
+// tokens fails.
+func (k EvmBankKeeper) SendCoinsFromModuleToAccount(ctx sdk.Context, senderModule string, recipientAddr sdk.AccAddress, amt sdk.Coins) error {
+	coins, err := splitAkavaCoins(amt)
+	if err != nil {
+		return err
+	}
+
+	if err := k.bk.SendCoinsFromModuleToAccount(ctx, senderModule, recipientAddr, coins); err != nil {
+		return err
+	}
+
+	return k.convertAkavaToUkava(ctx, recipientAddr)
+}
+
+// SendCoinsFromAccountToModule transfers akava coins from an AccAddress to
+// a ModuleAccount. It will panic if the module account does not exist.
+func (k EvmBankKeeper) SendCoinsFromAccountToModule(ctx sdk.Context, senderAddr sdk.AccAddress, recipientModule string, amt sdk.Coins) error {
+	coins, err := splitAkavaCoins(amt)
+	if err != nil {
+		return err
+	}
+
+	// convert 1 ukava to akava if sender account does not have enough akava to cover
+	akavaNeeded := coins.AmountOf(EvmDenom)
+	akavaBal := k.bk.GetBalance(ctx, senderAddr, EvmDenom).Amount
+	if akavaBal.LT(akavaNeeded) {
+		if err := k.convertOneUkavaToAkava(ctx, senderAddr); err != nil {
+			return err
+		}
+	}
+
+	return k.bk.SendCoinsFromAccountToModule(ctx, senderAddr, recipientModule, coins)
```

Comment:
note: this module should never have enough akava for a ukava so we shouldn't need to do `convertAkavaToUkava` here